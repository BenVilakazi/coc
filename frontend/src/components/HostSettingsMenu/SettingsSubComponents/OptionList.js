import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const propTypes = {
  listContent: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
  ).isRequired,
  state: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  onItemClick: PropTypes.func.isRequired,
  openText: PropTypes.string.isRequired,
  closedText: PropTypes.string.isRequired,
};

const OptionListButton = styled.button`
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);

  ${(props) =>
    props.state === 'open' &&
    css`
      color: var(--secondary-text-color);
      background-color: var(--secondary-background-color);
    `}

  ${(props) =>
    props.state === 'disabled' &&
    css`
      color: #dddddd;
      :hover {
        background-color: var(--primary-background-color);
      }
    `}

  display: flex;
  flex-direction: column;
  border: none;
  padding: 20px 25px;
  width: 100%;

  text-align: left;
  font-size: 1rem;
  font-weight: 700;

  cursor: pointer;

  :hover {
    background-color: var(--accent-background-color);
  }

  @media (min-width: 800px) {
    font-size: 1.5rem;
  }

  @media (min-width: 2000px) {
    font-size: 2.5rem;
  }

  @media (min-width: 3500px) {
    font-size: 4.5rem;
  }
`;

const OptionListItemButton = styled(OptionListButton)`
  font-size: 0.8rem;
  padding: 10px 50px;

  @media (min-width: 800px) {
    font-size: 1.3rem;
  }

  @media (min-width: 2000px) {
    font-size: 2.1rem;
  }

  @media (min-width: 3500px) {
    font-size: 4rem;
  }
`;

function OptionList({
  listContent,
  state,
  onClick: handleClick,
  onItemClick,
  openText,
  closedText,
}) {
  return (
    <div>
      <OptionListButton type="button" onClick={handleClick} state={state}>
        {state === 'open' ? openText : closedText}
      </OptionListButton>
      {state === 'open' &&
        listContent.map(({ text, value }) => (
          <OptionListItemButton
            type="button"
            key={text + value}
            onClick={() => onItemClick(value)}
          >
            {text}
          </OptionListItemButton>
        ))}
    </div>
  );
}

OptionList.propTypes = propTypes;

export default OptionList;
