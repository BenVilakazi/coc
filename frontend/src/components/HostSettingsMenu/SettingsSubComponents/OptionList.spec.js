import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import OptionList from './OptionList';

describe('OptionList', () => {
  describe('rendering', () => {
    it('renders a button with the given open text when open', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(
        screen.getByRole('button', { name: 'LIST IS OPEN' }),
      ).toBeInTheDocument();
    });

    it('renders a button with the given closed text when enabled', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="enabled"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(
        screen.getByRole('button', { name: 'LIST IS CLOSED' }),
      ).toBeInTheDocument();
    });

    it('does not render the contents of the list when enabled', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="enabled"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(
        screen.queryByRole('button', { name: 'foo' }),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByRole('button', { name: 'bar' }),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByRole('button', { name: 'baz' }),
      ).not.toBeInTheDocument();
    });

    it('renders a button with the given closed text when disabled', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="disabled"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(
        screen.getByRole('button', { name: 'LIST IS CLOSED' }),
      ).toBeInTheDocument();
    });

    it('does not render the contents of the list when disabled', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="disabled"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(
        screen.queryByRole('button', { name: 'foo' }),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByRole('button', { name: 'bar' }),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByRole('button', { name: 'baz' }),
      ).not.toBeInTheDocument();
    });

    it('renders the content of the list when open', () => {
      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(screen.getByRole('button', { name: 'foo' })).toBeInTheDocument();
      expect(screen.getByRole('button', { name: 'bar' })).toBeInTheDocument();
      expect(screen.getByRole('button', { name: 'baz' })).toBeInTheDocument();
    });

    it('matches the expected snapshot when open', () => {
      const tree = renderer
        .create(
          <OptionList
            listContent={[
              { text: 'foo', value: 'fooID' },
              { text: 'bar', value: 'barID' },
              { text: 'baz', value: 'bazID' },
            ]}
            state="open"
            onClick={() => {}}
            onItemClick={() => {}}
            openText="LIST IS OPEN"
            closedText="LIST IS CLOSED"
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('matches the expected snapshot when enabled', () => {
      const tree = renderer
        .create(
          <OptionList
            listContent={[
              { text: 'foo', value: 'fooID' },
              { text: 'bar', value: 'barID' },
              { text: 'baz', value: 'bazID' },
            ]}
            state="enabled"
            onClick={() => {}}
            onItemClick={() => {}}
            openText="LIST IS OPEN"
            closedText="LIST IS CLOSED"
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('matches the expected snapshot when disabled', () => {
      const tree = renderer
        .create(
          <OptionList
            listContent={[
              { text: 'foo', value: 'fooID' },
              { text: 'bar', value: 'barID' },
              { text: 'baz', value: 'bazID' },
            ]}
            state="disabled"
            onClick={() => {}}
            onItemClick={() => {}}
            openText="LIST IS OPEN"
            closedText="LIST IS CLOSED"
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('functionality', () => {
    it('does not call the onClick callback when the OptionList main button is not clicked', () => {
      const mockOnClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={mockOnClick}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(mockOnClick).not.toHaveBeenCalled();
    });

    it('calls the onClick callback when the OptionList main button is open and is clicked', () => {
      const mockOnClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={mockOnClick}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'LIST IS OPEN' }));
      expect(mockOnClick).toHaveBeenCalled();
    });

    it('calls the onClick callback when the OptionList main button is closed and is clicked', () => {
      const mockOnClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="enabled"
          onClick={mockOnClick}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'LIST IS CLOSED' }));
      expect(mockOnClick).toHaveBeenCalled();
    });

    it('does not call the onItemClick callback if none of the list item buttons are clicked', () => {
      const mockOnItemClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={mockOnItemClick}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(mockOnItemClick).not.toHaveBeenCalled();
    });

    it('does not call the onItemClick callback if none of the list item buttons are clicked but the main button is', () => {
      const mockOnItemClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={mockOnItemClick}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'LIST IS OPEN' }));
      expect(mockOnItemClick).not.toHaveBeenCalled();
    });

    it('calls the onItemClick callback once each time any of the list item buttons are clicked', () => {
      const mockOnItemClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={mockOnItemClick}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'foo' }));
      userEvent.click(screen.getByRole('button', { name: 'baz' }));
      expect(mockOnItemClick).toHaveBeenCalledTimes(2);
    });

    it('calls the onItemClick callback multiple times if a list item is clicked multiple times', () => {
      const mockOnItemClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={mockOnItemClick}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'foo' }));
      userEvent.click(screen.getByRole('button', { name: 'foo' }));
      userEvent.click(screen.getByRole('button', { name: 'baz' }));
      expect(mockOnItemClick).toHaveBeenCalledTimes(3);
    });

    it('calls the onItemClick callback with that respective value', () => {
      const mockOnItemClick = jest.fn();

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={mockOnItemClick}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      userEvent.click(screen.getByRole('button', { name: 'foo' }));
      expect(mockOnItemClick).toHaveBeenCalledWith('fooID');
    });
  });

  describe('propTypes', () => {
    it('logs an error when attempting to render without the listContent prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      expect(() => {
        render(
          <OptionList
            state="open"
            onClick={() => {}}
            onItemClick={() => {}}
            openText="LIST IS OPEN"
            closedText="LIST IS CLOSED"
          />,
        );
      }).toThrow();
      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the listContent prop as a non array', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      expect(() => {
        render(
          <OptionList
            listContent="foo"
            state="open"
            onClick={() => {}}
            onItemClick={() => {}}
            openText="LIST IS OPEN"
            closedText="LIST IS CLOSED"
          />,
        );
      }).toThrow();
      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the listContent prop as an array of non objects', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[true, false, true]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when the listContent prop objects lack a text property', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[{ value: 'foo' }, { value: 'bar' }, { value: 'baz' }]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when the listContent prop objects text property is not a string', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: true, value: 'foo' },
            { text: true, value: 'bar' },
            { text: true, value: 'baz' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when the listContent prop objects lack a value property', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[{ text: 'foo' }, { text: 'bar' }, { text: 'baz' }]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when the listContent prop objects value property is not a string', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: true },
            { text: 'bar', value: true },
            { text: 'baz', value: true },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render without the state prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the state prop as a non string', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state={false}
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render without the onClick prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the onClick prop as a non function', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={false}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render without the onItemClick prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the onItemClick prop as a non function', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={false}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render without the openText prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onItemClick={() => {}}
          onClick={() => {}}
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the openText prop as a non string', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText={false}
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render without the closedText prop', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onItemClick={() => {}}
          onClick={() => {}}
          openText="LIST IS OPEN"
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('logs an error when attempting to render the closedText prop as a non string', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText={false}
        />,
      );

      expect(consoleSpy).toHaveBeenCalled();
    });

    it('does not log any error when all of the needed props are provided', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(
        <OptionList
          listContent={[
            { text: 'foo', value: 'fooID' },
            { text: 'bar', value: 'barID' },
            { text: 'baz', value: 'bazID' },
          ]}
          state="open"
          onClick={() => {}}
          onItemClick={() => {}}
          openText="LIST IS OPEN"
          closedText="LIST IS CLOSED"
        />,
      );

      expect(consoleSpy).not.toHaveBeenCalled();
    });
  });
});
