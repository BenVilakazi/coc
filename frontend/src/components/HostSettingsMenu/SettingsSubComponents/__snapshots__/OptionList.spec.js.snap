// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionList rendering matches the expected snapshot when disabled 1`] = `
.c0 {
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);
  color: #dddddd;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: none;
  padding: 20px 25px;
  width: 100%;
  text-align: left;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--primary-background-color);
}

.c0:hover {
  background-color: var(--accent-background-color);
}

@media (min-width:800px) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:2000px) {
  .c0 {
    font-size: 2.5rem;
  }
}

@media (min-width:3500px) {
  .c0 {
    font-size: 4.5rem;
  }
}

<div>
  <button
    className="c0"
    onClick={[Function]}
    type="button"
  >
    LIST IS CLOSED
  </button>
</div>
`;

exports[`OptionList rendering matches the expected snapshot when enabled 1`] = `
.c0 {
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: none;
  padding: 20px 25px;
  width: 100%;
  text-align: left;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--accent-background-color);
}

@media (min-width:800px) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:2000px) {
  .c0 {
    font-size: 2.5rem;
  }
}

@media (min-width:3500px) {
  .c0 {
    font-size: 4.5rem;
  }
}

<div>
  <button
    className="c0"
    onClick={[Function]}
    type="button"
  >
    LIST IS CLOSED
  </button>
</div>
`;

exports[`OptionList rendering matches the expected snapshot when open 1`] = `
.c0 {
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);
  color: var(--secondary-text-color);
  background-color: var(--secondary-background-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: none;
  padding: 20px 25px;
  width: 100%;
  text-align: left;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--accent-background-color);
}

.c1 {
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: none;
  padding: 20px 25px;
  width: 100%;
  text-align: left;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c1:hover {
  background-color: var(--accent-background-color);
}

.c2 {
  font-size: 0.8rem;
  padding: 10px 50px;
}

@media (min-width:800px) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:2000px) {
  .c0 {
    font-size: 2.5rem;
  }
}

@media (min-width:3500px) {
  .c0 {
    font-size: 4.5rem;
  }
}

@media (min-width:800px) {
  .c1 {
    font-size: 1.5rem;
  }
}

@media (min-width:2000px) {
  .c1 {
    font-size: 2.5rem;
  }
}

@media (min-width:3500px) {
  .c1 {
    font-size: 4.5rem;
  }
}

@media (min-width:800px) {
  .c2 {
    font-size: 1.3rem;
  }
}

@media (min-width:2000px) {
  .c2 {
    font-size: 2.1rem;
  }
}

@media (min-width:3500px) {
  .c2 {
    font-size: 4rem;
  }
}

<div>
  <button
    className="c0"
    onClick={[Function]}
    type="button"
  >
    LIST IS OPEN
  </button>
  <button
    className="c1 c2"
    onClick={[Function]}
    type="button"
  >
    foo
  </button>
  <button
    className="c1 c2"
    onClick={[Function]}
    type="button"
  >
    bar
  </button>
  <button
    className="c1 c2"
    onClick={[Function]}
    type="button"
  >
    baz
  </button>
</div>
`;
